notebooks/problem-sets/quantum_enigma_005.ipynb

```diff
@@ -0,0 +1,371 @@
+    "</details>\n",
+    "\n",
+    "<details>\n",
+    "    <summary>Hint 2</summary>\n",
+    "    The SWAP gate can be taken off if the subsequent operations are adjusted between the two qubits.\n",
+    "</details>\n",
+    "\n",
+    "<details>\n",
+    "    <summary>Hint 3</summary>\n",
+    "    If a CNOT has the same control and target as another CNOT for which two NOT gates are applied before and after the control qubit, this can be simplified to a single NOT gate on the target qubit of the CNOT as a NOT gate is applied to the target whether the control qubit is initially in state 0 or 1.<br><br>\n",
+    "    This simplification is possible because a NOT gate is applied to the target whether the control qubit is initially in state 0 or 1.<br><br>\n",
+    "    <img src=\"images/enigma001_problem3.png\">\n",
+    "</details>\n",
+    "\n",
+    "<details>\n",
+    "    <summary>Hint 4</summary>\n",
+    "    The circuit can be simplified until only three gates remain in the algorithm.\n",
+    "</details>\n",
+    "<!-- ::: -->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from qiskit import QuantumCircuit\n",
+    "\n",
+    "qc = QuantumCircuit(2)\n",
+    "\n",
+    "qc.h(0)\n",
+    "qc.x(1)\n",
+    "qc.h(1)\n",
+    "qc.h(0)\n",
+    "\n",
+    "qc.draw(output='mpl')"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<!-- ::: q-block.exercise -->\n",
+    "\n",
+    "### Problem 2\n",
+    "\n",
+    "<!-- ::: q-quiz(goal=\"enigma-005-quiz-2\") -->\n",
+    "\n",
+    "<!-- ::: .question -->\n",
+    "\n",
+    "Which statement is true if you simplify the circuit for the following situation representing when Constantin does not repaint the statues.\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: .option(correct) -->\n",
+    "\n",
+    "1. No gate is left on $q_0$ and its state is 0.\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: .option -->\n",
+    "\n",
+    "2. No gate is left on $q_1$ and its state is 1.\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: .option -->\n",
+    "\n",
+    "3. Nothing changes.\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: .option -->\n",
+    "\n",
+    "4. Only one H gate is left on $q_0$ and it is in a state superposition.\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: -->"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Problem 3 + 4\n",
+    "\n",
+    "Write a circuit that would allow you to randomly determine which oracle is used i.e. which vandal used which scenario.\n",
+    "\n",
+    "Use the following legend:\n",
+    "\n",
+    "1. Qubit 2 determines which vandal is involved: 0 is Constantin, 1 is Equilibria\n",
```

Comment:
Add the following before Qubit 2

Qubit 0 is the park: O is Ozis and 1 is Ida
Qubit 1 is the colour: 0 is orange and 1 in indigo